ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -418,6 +411,20 @@ func (scbuilder *SiteConfigBuilder) getExtraManifest(dataMap map[string]interfac
 		}
 	}
 
+	//filer CRs
+	dataMap, err = filterExtraManifests(dataMap, clusterSpec.ExtraManifests.Filter)
+	if err != nil {
+		log.Printf("could not filter %s.%s %s\n", clusterSpec.ClusterName, clusterSpec.ExtraManifestPath, err)
+		return dataMap, err
+	}
+
+	// merge the pre-defined manifests
+	dataMap, err = MergeManifests(dataMap, doNotMerge)
+	if err != nil {
+		log.Printf("Error could not merge extra-manifest %s.%s %s\n", clusterSpec.ClusterName, clusterSpec.ExtraManifestPath, err)
+		return dataMap, err
+	}
+
```

Comment:
@tliu2021 could take a look at this when you get a chance? The new filter function needs to have the full list before you can merge.

Reply:
Actually moving it causes tests to fail. Need your help with this

Reply:
By moving the merge function down here you're implicitly adding all the user manifests to the merged MachineConfig.  If you set the `doNotMerge[file]=true` inside the user-supplied extra manifests loop (line 388) I bet it will work again.

Reply:
Yep tried that. Some kind of path issues it seems (tests pass if I move it back)...investigating.

Reply:
Had to revert the test file used to assert `Pre-defined extra-manifest cannot be over written`. Looking good now.